dev/com.ibm.ws.st.core/src/com/ibm/ws/st/core/internal/ConsoleMonitorThread.java

```diff
@@ -126,17 +131,31 @@ public void run() {
                     synchronized (serverStateSyncObj) {
                         wsBehaviour.stopImpl();
                     }
-                }
-                // At this point we know the server is not started based on the status from
-                // WebSphereRuntime.isServerStarted() called in the first "if" block.
-                // If the server state is started we should terminate the launch which will change
-                // the server to STOPPED state.
-                else if (state == IServer.STATE_STARTED) {
+                } else if (state == IServer.STATE_STARTED) {
```

Comment:
Because of the newly introduced `if (serverStatus == 0) { … } else if { … }`, this line can never be accessed in the first cycle and needs to wait for `Thread.sleep(POLLING_DELAY);` from line 165. Is this intended?

Reply:
The old code is:
```
// server process is running, but we aren't tracking the state yet so need to add the process listener to listen for console messages
                if (wsRuntime != null && wsServer.getUserDirectory() != null && wsRuntime.isServerStarted(wsServer.getServerInfo(), null)) {
```
Which means that it will not go through unless the runtime is not null, the user directory is not null and the server is started.
The new code is:
```
if (wsRuntime != null && wsServer.getUserDirectory() != null) {
                    serverStatus = wsRuntime.getServerStatus(wsServer.getServerInfo(), 20f, new NullProgressMonitor());
                }
                if (serverStatus == 0) {
```
Which means it will not go through unless the runtime is not null, the user directory is not null and the server is started (serverStatus == 0).  I just need the serverStatus elsewhere so moved it out but it is the same as:
```
 if (wsRuntime != null && wsServer.getUserDirectory() != null && wsRuntime.getServerStatus(wsServer.getServerInfo(), 20f, new NullProgressMonitor() == 0 ) {
```

Reply:
The old code was:
```
                 if (wsRuntime != null && wsServer.getUserDirectory() != null && wsRuntime.isServerStarted(wsServer.getServerInfo(), null)) {
```
Which means it will not enter unless the runtime is not null, the user directory is not null and the server is running.

The new code is:
```
                int serverStatus = -2; // Set to some value not returned by WebSphereRuntime.getServerStatus so can differentiate
                if (wsRuntime != null && wsServer.getUserDirectory() != null) {
                    // getServerStatus returns -1 if timeout; 0 if the server is running; 1 if the server is stopped; 2 if the status is unknown.
                    serverStatus = wsRuntime.getServerStatus(wsServer.getServerInfo(), 20f, new NullProgressMonitor());
                }
                if (serverStatus == 0) {
```
Which means it will not enter unless the runtime is not null, the user directory is not null and the server is running.  I just needed the serverStatus elsewhere so moved it out.  It is the same as:
```
                 if (wsRuntime != null && wsServer.getUserDirectory() != null && wsRuntime.getServerStatus(wsServer.getServerInfo(), 20f, new NullProgressMonitor())) {
```